docs/writing-a-profile.md

```diff
@@ -0,0 +1,166 @@
+# Writing a profile
+
+**Per endpoint, `hafas-client` has an endpoint-specific customisation called *profile*** which may for example do the following:
+
+- handle the additional requirements of the endpoint (e.g. authentication),
+- extract additional information from the data provided by the endpoint,
+- guard against triggering bugs of certain endpoints (e.g. time limits).
+
+This guide is about writing such a profile. If you just want to use an already supported endpoint, refer to the [API documentation](readme.md) instead.
+
+*Note*: **If you get stuck, ask for help by [creating an issue](https://github.com/derhuerst/hafas-client/issues/new)!** I want to help people expand the scope of this library.
```

Comment:
Optional: Change `I want to` to `We want to`, if you want.

Reply:
Ah. I've written the guide in the "I" form for now, but I'm happy to change to "we".